opm/simulators/timestepping/TimeStepControl.cpp

```diff
@@ -141,17 +142,23 @@ namespace Opm
         errors_[ 2 ] = error;
         for( int i=0; i<2; ++i ) {
             assert(std::isfinite(errors_[i]));
-            assert(errors_[i]>0);
+            assert(errors_[i]>=0);
         }
 
-        if( error > tol_ )
+        if( errors_[2] > tol_ )
         {
             // adjust dt by given tolerance
             const double newDt = dt * tol_ / error;
             if( verbose_ )
                 std::cout << "Computed step size (tol): " << unit::convert::to( newDt, unit::day ) << " (days)" << std::endl;
             return newDt;
         }
+        else if (errors_[1] * errors_[2] == 0.)
+        {
+            if ( verbose_ )
+                std::cout << "The solution between time step does not change, there is no time step constraint from the PID time step control " << std::endl;
+            return std::numeric_limits<double>::max();
```

Comment:
I am wondering whether the return value would ever be used directly or only as upper bound when using the minimum (like below or in AdaptiveTimeSteppingEbos.hpp). Assuming the latter this seems ok.